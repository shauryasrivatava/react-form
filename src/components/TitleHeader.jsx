import React from 'react';
import PropTypes from 'prop-types';

const TitleHeader = (props) => {
    return <h1 className={props.className}>{props.title}</h1>
}

TitleHeader.propTypes = {
    className: PropTypes.string,
    title: PropTypes.string
};

TitleHeader.defaultProps = {
    className: 'text-center',
    title: 'Hello World'
};

export default TitleHeader;