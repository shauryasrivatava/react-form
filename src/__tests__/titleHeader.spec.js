import renderer from 'react-test-renderer';
import TitleHeader from '../components/TitleHeader';
import React from 'react';
import { mount } from 'enzyme';
import enzymeConfig from '../../enzymeConfig'

describe('<TitleHeader />', function(){

    it('should render correctly', function(){
        var title = "Hello World";
        const component = renderer.create(<TitleHeader title={title}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should have the correct title prop', function(){
        var title = "New Title";
        const component = mount(<TitleHeader title={title} />);
        expect(component.find('h1').text()).toEqual('New Title');
    });
})