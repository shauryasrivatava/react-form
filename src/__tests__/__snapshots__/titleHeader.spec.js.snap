// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleHeader /> should render correctly 1`] = `
<h1
  className="text-center"
>
  Hello World
</h1>
`;
